### PR TITLE
chore: (litmus-portal) Final build bug fixes, refactoring and imporvements

### PR DESCRIPTION
**Pull request by arkajyotiMukherjee**
## Proposed changes
- refactored directory structure for frontend
- browse workflow table fixes and header notification fixes
- settings and project selection fixes
- project switching fixes, routing changes for settings and disabling buttons for release
- routes and breadcrumbs fixed
- tab switching improved with redux

Summarize your changes here to communicate with the maintainers and make sure to put the link of that issue

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [ ] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

File: litmus-portal/frontend/src/components/Header/index.tsx [hunk 1]
```diff
@@ -57,12 +57,12 @@ const Header: React.FC = () => {
...
             user.updateUserDetails({
               selectedProjectID,
-              userRole: project.name,
+              userRole: member.role,
               selectedProjectName: member.role,
             });
             setSelectedProjectDetails({
               selectedProjectID,
-              selectedProjectName: project.name,
+              selectedProjectName: member.role,
```
Comment (ishangupta-ds) [inline litmus-portal/frontend/src/components/Header/index.tsx:65]: why the `selectedProjectName` variable has been assigned `member.role` data?

**Review by ishangupta-ds — COMMENTED** [review]
@arkajyotiMukherjee

**Review by amityt — DISMISSED** [review]
LGTM

**Review by Saranya-jena — APPROVED** [review]

**Review by ishangupta-ds — DISMISSED** [review]
LGTM.

**Comment by rajdas98** [comment]
Thanks for this PR. Can you check the codacy issue? It is not showing here. But you can go to this link- https://app.codacy.com/gh/litmuschaos/litmus/pullRequest?prid=6157531

**Review by S-ayanide — DISMISSED** [review]

**Comment by arkajyotiMukherjee** [comment]
> Thanks for this PR. Can you check the codacy issue? It is not showing here. But you can go to this link- https://app.codacy.com/gh/litmuschaos/litmus/pullRequest?prid=6157531

looking into it

**Comment by arkajyotiMukherjee** [comment]
> Thanks for this PR. Can you check the codacy issue? It is not showing here. But you can go to this link- https://app.codacy.com/gh/litmuschaos/litmus/pullRequest?prid=6157531

few of the codacy fixes are done...rest of them are because codacy can't understand/parse typescript interfaces
@rajdas98

**Review by S-ayanide — APPROVED** [review]

**Review by amityt — APPROVED** [review]

**Review by Saranya-jena — APPROVED** [review]

**Pull request merged**